SQL/optimisations_2017-02-19.sql

```diff
@@ -0,0 +1,191 @@
+, ADD COLUMN `server_port` SMALLINT UNSIGNED NOT NULL AFTER `server_ip`
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
+ SET `server_ip` = COALESCE(NULLIF(INET_ATON(SUBSTRING_INDEX(`serverip`, ':', 1)), ''), INET_ATON('0.0.0.0'))
+, `server_port` = IF(`serverip` LIKE '%:_%', CAST(SUBSTRING_INDEX(`serverip`, ':', -1) AS UNSIGNED), '0')
+, `ipTEMP` = COALESCE(NULLIF(INET_ATON(SUBSTRING_INDEX(`ip`, ':', 1)), ''), INET_ATON('0.0.0.0'));
+SET SQL_SAFE_UPDATES = 1;
+ALTER TABLE `feedback`.`connection_log`
+ DROP COLUMN `ip`
+, DROP COLUMN `serverip`
+, CHANGE COLUMN `ipTEMP` `ip` INT(10) UNSIGNED NOT NULL;
+COMMIT;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`death`
+ CHANGE COLUMN `pod` `pod` VARCHAR(50) NOT NULL
+, CHANGE COLUMN `coord` `coord` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `mapname` `mapname` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `job` `job` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `special` `special` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `name` `name` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `byondkey` `byondkey` VARCHAR(32) NOT NULL
+, CHANGE COLUMN `laname` `laname` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `lakey` `lakey` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `gender` `gender` ENUM('neuter', 'male', 'female', 'plural') NOT NULL
+, CHANGE COLUMN `bruteloss` `bruteloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `brainloss` `brainloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `fireloss` `fireloss` SMALLINT UNSIGNED NOT NULL
+, CHANGE COLUMN `oxyloss` `oxyloss` SMALLINT UNSIGNED NOT NULL
+, ADD COLUMN `server_ip` INT UNSIGNED NOT NULL AFTER `server`
+, ADD COLUMN `server_port` SMALLINT UNSIGNED NOT NULL AFTER `server_ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`ban`
+ SET `server_ip` = COALESCE(NULLIF(INET_ATON(SUBSTRING_INDEX(`server`, ':', 1)), ''), INET_ATON('0.0.0.0'))
+, `server_port` = IF(`server` LIKE '%:_%', CAST(SUBSTRING_INDEX(`server`, ':', -1) AS UNSIGNED), '0');
+SET SQL_SAFE_UPDATES = 1;
+ALTER TABLE `feedback`.`death`
+ DROP COLUMN `server`;
+COMMIT;
+
+ALTER TABLE `feedback`.`library`
+ CHANGE COLUMN `category` `category` ENUM('Any', 'Fiction', 'Non-Fiction', 'Adult', 'Reference', 'Religion') NOT NULL
+, CHANGE COLUMN `ckey` `ckey` VARCHAR(32) NOT NULL DEFAULT 'LEGACY'
+, CHANGE COLUMN `datetime` `datetime` DATETIME NOT NULL
+, CHANGE COLUMN `deleted` `deleted` TINYINT(1) UNSIGNED NULL DEFAULT NULL;
+
+ALTER TABLE `feedback`.`messages`
+ CHANGE COLUMN `type` `type` ENUM('memo', 'message', 'message sent', 'note', 'watchlist entry') NOT NULL
+, CHANGE COLUMN `text` `text` VARCHAR(2048) NOT NULL
+, CHANGE COLUMN `secret` `secret` TINYINT(1) UNSIGNED NOT NULL;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`player`
+ ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`player`
+ SET `ip` = IF(`ip` IS NULL OR `ip` = '','0',`ip`);
+UPDATE `feedback`.`player`
+ SET `ipTEMP` = INET_ATON(`ip`);
```

Comment:
Should maybe use the single query method you figured out for the ips in other tables on this one too.